### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by dgiglia**
Any comments very much appreciated. Thanks :)

### CODE REVIEW

File: app/controllers/admin/videos_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @video = Video.new
+  end
+  
+  def create
+    @video = Video.create(video_params)
+    if @video.save
+      flash['success'] = "#{@video.title} has been added."
+      redirect_to new_admin_video_path
+    else
+      flash['danger'] = "Video has not been added."
```
Comment (falonofthetower) [inline app/controllers/admin/videos_controller.rb]: this should be `flash.now['danger']` you want the flash _now_ when you render but his flash is fast enough with a redirect.

File: app/views/videos/show.html.haml [hunk 2]
```diff
@@ -9,7 +9,7 @@
           %span Rating: #{@video.average_rating} / 5
         %p= @video.description
         .actions
-          %a.btn.btn-primary(href="") Watch Now
+          %a.btn.btn-primary(href="#{@video.video_url}") Watch Now
```
Comment (falonofthetower) [inline app/views/videos/show.html.haml]: This could be a `link_to`

File: spec/controllers/admin/videos_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Admin::VideosController do
+  it {is_expected.to use_before_action(:require_user)}
+  
+  describe "GET new" do
+    before {set_current_user}
```
Comment (falonofthetower) [inline spec/controllers/admin/videos_controller_spec.rb]: This before is the start of a poorly formed block. You set the before (which should be necessary for every test in the block) and then override it with `set_current_admin` and then don't really use it in either of the final tests. There are a couple of ways to restructure this. Either drop the before block and simply write out the tests in full or use a context for each type of user. In this case (especially with your `it_behaves_like`) I would probably simply shift `set_current_user` to line 15. If it were something larger I would consider a context.

Regardless the most important thing is that every line matters. Using a `before` applies to every test in the block and should therefore be required for every test--if it doesn't don't use it or restructure the offending test out. Use techniques such as `before` to trim away fat, but don't allow their abuse to muddy the waters.

File: spec/controllers/admin/videos_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Admin::VideosController do
+  it {is_expected.to use_before_action(:require_user)}
```
Comment (falonofthetower) [inline spec/controllers/admin/videos_controller_spec.rb:4]: I wouldn't do this here. I'm not sure it actually works like this. Even if it did what you really want to do is test the results (that the user was redirect) not the implementation (that `require_user`) was used.

File: spec/controllers/admin/videos_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      let(:action) {get :new}
+    end
+    
+    it_behaves_like "require admin" do
+      let(:action) {get :new}
+    end
+  end
+  
+  describe "POST create" do
+    before {set_current_user}
```
Comment (falonofthetower) [inline spec/controllers/admin/videos_controller_spec.rb]: Same issue here. In fact your later `set_current_admin` overrides this in both cases.

File: spec/controllers/admin/videos_controller_spec.rb (hunk 4 above)
Comment (dgiglia) [inline spec/controllers/admin/videos_controller_spec.rb:4]: I have this in most of my specs. The shoulda matcher docs say:
`use_before_action` tests that a `before_action` callback is defined in your controller. 

So basically just insurance that it's called. I later test for the redirect. Is it pointless to have these at all?

File: spec/controllers/admin/videos_controller_spec.rb (hunk 3 above)
Comment (dgiglia) [inline spec/controllers/admin/videos_controller_spec.rb]: Thanks. Yeah, that was pointless for me to use the before. Completely understand. Thanks for calling me on it! :)

**Pull request merged**